lib/herd/zip/import.rb

```
@@ -0,0 +1,64 @@
+module Herd
+  module Zip
+    class Import < Base
+      attr_accessor :zip_path
+
+      def self.import(zip_path)
+        new(zip_path).import
+      end
+
+      def initialize(zip_path)
+        @zip_path = zip_path
+      end
+
+      def import(path=nil)
+        @zip_path = path if path
+        zip_data = open(zip_path)
+
+        assets=[]
+
+        ::Zip::File.open(zip_data) do |zip|
+          zip.each do |entry|
+            next if entry.name =~ /\.DS_Store|__MACOSX|(^|\/)\._/
+            # FIXME: mus b better way
+            next unless %w(.jpg .gif .png).include? File.extname(entry.name).downcase
+
+            parts = entry.name.split '/'
+            parts.shift if parts.first.match Regexp.new('seed', 'g')
+
+            asset_file = parts.pop
+            assetable_slug = parts.pop
+            klass = class_from_path parts.join('/')
+            raise "no class Herd::#{model}" unless klass
```

I don't think you'll ever throw this exception because trying to constantize the string should throw an error if the class doesn't exist.